### PR TITLE
Convert entity panels to pages

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Converted units / machines / app panels to pages
- Redirect to 404 if trying to access a model via refresh or direct URL that is not accessible
- Update and simplify `<ChipGroup />`
- Uncouple entity metadata into a separate component from `<InfoPanel>`
- Fix bouncing sidebar
- Updated Breadcrumbs to be dynamic to different entity types

## QA

- Run the app using `yarn start` or demo link
- Click around and make sure everything works as expected

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1624,
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1621,
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1623,
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1625,
Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1628,
Fixes https://github.com/canonical-web-and-design/jaas-dashboard/issues/885

### CODE REVIEW

File: src/components/ChipGroup/ChipGroup.tsx [hunk 1]
```diff
@@ -1,30 +1,67 @@
...
+      {numberOfChips > 0 && (
+        <div className="chip-group">
+          {labelType && (
+            <strong className="chip-group__descriptor">
+              {`${numberOfChips} ${labelType}`}
+            </strong>
+          )}
+          {Object.entries(chips).map(([label, count]) => {
+            return (
+              // @ts-ignore Can't seem to apply type Chip to [label,count] above?
```
Comment (barrymcgee) [inline src/components/ChipGroup/ChipGroup.tsx:52]: @hatched Let's chat about this.

File: src/layout/BaseLayout/BaseLayout.test.tsx [hunk 2]
```diff
@@ -45,7 +47,10 @@ describe("Base Layout", () => {
   });
 
   it("should collapse the sidebar on entity details pages", () => {
-    const store = mockStore(dataDump);
+    const clonedDump = cloneDeep(dataDump);
+    // @ts-ignore sideNavCollapsed doesn't appear on type '{ userMenuActive: boolean; }'?
```
Comment (barrymcgee) [inline src/layout/BaseLayout/BaseLayout.test.tsx:51]: @hatched Let's talk about this.

**Comment by long-chung** [comment]
Really great work. 
One minor comment, could you adopt the same styling as Ant did for the models segment control please. i.e all the segments should be the same width as the one with the longest text.
Screenshot 2021-03-04 at 14 27 21
Screenshot 2021-03-04 at 14 27 37

File: src/pages/EntityDetails/counts.js [hunk 3]
```diff
@@ -31,5 +25,32 @@ export const generateUnitSecondaryCounts = (modelStatusData) => {
...
+      break;
+    case "machines":
+      chips = generateSecondaryCounts(
+        modelStatusData,
+        "machines",
+        "agent-status"
+      );
+      break;
+    case "relations":
+      return null;
```
Comment (hatched) [inline src/pages/EntityDetails/counts.js:53]: this should probably just have a `break;` here and then chips will be returned as null.

File: src/pages/EntityDetails/EntityDetails.js [hunk 4]
```diff
@@ -29,8 +34,37 @@ import { fetchModelStatus } from "juju/actions";
...
+  }
+}
+
+const EntityDetails = ({ type, children }) => {
+  const modelStatusData = useModelStatus();
+  const { userName, modelName } = useParams();
+  const history = useHistory();
+
+  // Check if model exists in modelStatusData, else 404
+  if (modelName !== modelStatusData?.model?.name) {
```
Comment (hatched) [inline src/pages/EntityDetails/EntityDetails.js:52]: This makes it impossible to access a model before the application has had a chance to actually load the model data. To reproduce view a model then an app then refresh, it'll 404.

**Review by hatched — CHANGES_REQUESTED** [review]

**Review by hatched — APPROVED** [review]
Thanks for all this work!

**Pull request merged**